src/plugins/memdump/memdump.cpp

```diff
@@ -277,6 +277,11 @@ bool dump_memory_region(
     fclose(fp);
 
     chk_str = g_checksum_get_string(checksum);
+    if (plugin->dumped_hashes.find(chk_str) != plugin->dumped_hashes.end()) {
+        // We have already dumped this memory region.
+        goto done;
+    }
+    plugin->dumped_hashes.insert(chk_str);
```

Comment:
This is some voodoo magic as it's auto converting a `const gchar*` to `std::string`.

Reply:
If I'm not mistaken insert will copy the argument and so will create a `std::string` out of `const gchar*`

Reply:
but I will change this

Reply:
It seems valid to me. Glib docs say that  `gchar` "Corresponds to the standard C char type", so `const gchar*` is literally `const char *`

Reply:
I've fixed that. I agree that it's better to be explicit.